runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -243,26 +250,32 @@ class MM_ObjectAllocationAPI
 				}
 
 				/* Initialize the object */
-				objectHeader->clazz = (j9objectclass_t)(UDATA)arrayClass;
-				objectHeader->size = size;
-
-				if (initializeSlots) {
-					memset(objectHeader + 1, 0, dataSize);
+				if (J9VMTHREAD_COMPRESS_OBJECT_REFERENCES(currentThread)) {
+					J9IndexableObjectContiguousCompressed *header = (J9IndexableObjectContiguousCompressed*)objectHeader;
+					header->clazz = (j9objectclass_t)(UDATA)arrayClass;
+					header->size = size;
+					if (initializeSlots) {
+						memset(header + 1, 0, dataSize);
+					}
+				} else {
+					J9IndexableObjectContiguousFull *header = (J9IndexableObjectContiguousFull*)objectHeader;
+					header->clazz = (j9objectclass_t)(UDATA)arrayClass;
+					header->size = size;
+					if (initializeSlots) {
+						memset(header + 1, 0, dataSize);
+					}
 				}
 				if (memoryBarrier) {
 					VM_AtomicSupport::writeBarrier();
 				}
-				instance = (j9object_t) objectHeader;
+				instance = objectHeader;
 			}
 		} else {
-			/* Discontiguous Array */
-#if J9SIZEOF_J9IndexableObjectDiscontiguous <= J9_GC_MINIMUM_OBJECT_SIZE
-
-			/* Calculate the object size */
+			/* Zero-length array is discontiguous)Discontiguous Array - assume minimum object size */
 			UDATA allocateSize = J9_GC_MINIMUM_OBJECT_SIZE;
 
 			/* Allocate the memory */
-			J9IndexableObjectDiscontiguous *objectHeader = NULL;
+			void *objectHeader = NULL;
```

Comment:
Why it is not symmetric with Contiguous Array case? Here it is a `void *` and there was a `j9object_t `

Reply:
Also fixed.